### PR TITLE
Updates to Governance.md by Brian Lynn

### PR DESCRIPTION
**Pull request by eteridvalishvili**

### CODE REVIEW

File: GOVERNANCE.md [hunk 1]
```diff
@@ -54,3 +54,204 @@ Inspired by [ANSI’s Essential Requirements for Due Process](https://share.ansi
...
+* Minor Production releases shall be reported in brief to the CRWG and the SWG, 
+* A roadmap of anticipated minor releases shall be reported by the maintainers to the CRWG based on PRs that are in process.
+
+## 9.4  Production Patch Release Scheduling and Approvals
+
+* Patch releases to correct defects without releasing new functionality may be scheduled by the maintainers based on the presence of approves defect correction PRs, or other non-functional PRs (e.g. security remediations).
+*  Patch releases require the approval of one maintainer
+*  Patch releases shall be reported to the CRWG.
+
+ ## 9.5 Summary of Release Approval Requirements
```
Comment (chrisisla) [inline GOVERNANCE.md:248]: similar comment on the terms used in the table for Major Release, Minor Release etc. Do we want to make these consistent with the rest of this section where we refer to Major Production Release, Minor Production Release, Production Patch Release?

**Review by chrisisla — CHANGES_REQUESTED** [review]

File: GOVERNANCE.md [hunk 2]
```diff
@@ -54,3 +54,204 @@ Inspired by [ANSI’s Essential Requirements for Due Process](https://share.ansi
...
+* No major release will be planned/scheduled  (decision and content) without formal approval at a meeting of the SWG
+  * *Rationale:*  Designation of a major release is an important decision that requires transparency and control
+ 
+* The intention is that major releases shall be planned and reviewed at the SWG at least 3 months ahead of the anticipated release date.  
+  * *Rationale:*  Giving the community advance warning of major changes will help CDM users plan for how they will use CDM and avoid major surprises.  It will also help CDM developers plan their own changes
+    
+* It is anticipated that for at least the next several years (say 4-5) at least one major release will be planned each year.
+  * *Rationale:* we anticipate that there will be an accumulation of desired changes that cannot be accommodated within a minor release and we wish to ensure that these can be addressed without undue delay
+    
+*Any addition to the scope/contents of (or technical change to) a planned major release requires SWG approval
```
[the remark this answers is not in the document]
Comment (brianlynn2) [reply GOVERNANCE.md:96]: agreed

File: GOVERNANCE.md [hunk 3]
```diff
@@ -54,3 +54,204 @@ Inspired by [ANSI’s Essential Requirements for Due Process](https://share.ansi
...
+* Rules
+  * The specific rules we want to define and enforce to meet the principles
+* Evaluation methods
+  * How we want to ensure that the rules are evaluated and enforced during development
+  * This includes development processes (e.g. review and approval) as well as automated tooling (e.g. regression test cases)
+
+## 7.1 Change Control Principles
+
+* We are trying to ensure rapid, smooth, and predictable evolution of the model by controlling when and how breaking changes are introduced
+  * We want to allow changes where needed, with a defined process make those changes, to meet evolved and improved understanding of the business and technical requirements.
```
[the remark this answers is not in the document]
Comment (brianlynn2) [reply GOVERNANCE.md:144]: agreed

File: GOVERNANCE.md [hunk 4]
```diff
@@ -54,3 +54,204 @@ Inspired by [ANSI’s Essential Requirements for Due Process](https://share.ansi
...
+  * We want to give ourselves some freedom to make changes more easily when there are newly introduced components/structures that may not be fully mature, but we don’t want to spend a lot of effort on planning for that.  We will do this using the defect approval guidelines, giving some scope for correcting recently introduced changes.
```
[the remark this answers is not in the document]
Comment (brianlynn2) [reply GOVERNANCE.md:145]: We changed to "bug fix" to narrow the scope of bug fixes to eliminate redesigns and enhancements, per Minesh recommendation.   I agree that this should be consistent ... recommend changing the last sentence to:

We will do this using the pull request approval guidelines for bug fixes, giving some scope for correcting recently introduced changes.

File: GOVERNANCE.md [hunk 5]
```diff
@@ -54,3 +54,204 @@ Inspired by [ANSI’s Essential Requirements for Due Process](https://share.ansi
...
+    * Change to the DSL that results in any existing expression becoming invalid is prohibited
+    * All validations that pass in version M.N should also pass in version M.N+1
+  * Function signatures may not be changed in such a way as to invalidate previous callers (e.g addition of new mandatory parameters, or removal/change of existing parameters.)
+    * Change to the DSL that results in change to any of the generated code's public interfaces is prohibited
+  * Test cases that passed in a prior version shall continue to work.
+  * We allow some minor exceptions to these rules for newly introduced functionality that may not be fully formed, as part of the PR process for defect corrections
+  * Functionality shall not be removed between major versions without advance notice
+
+
+Please note that full, bidirectional interoperability between minor versions is not required.  If an application uses functionality in version x.y, it does not need to fully interoperate with version x.y-1, assuming that the older version does not include that functionality.  However, if an application uses functionality found in version x.y, it should be able to interoperate with version x.y+1.
```
[the remark this answers is not in the document]
Comment (brianlynn2) [reply GOVERNANCE.md:165]: agreed

File: GOVERNANCE.md [hunk 6]
```diff
@@ -54,3 +54,204 @@ Inspired by [ANSI’s Essential Requirements for Due Process](https://share.ansi
...
+
+
+## 8.0 Pull Request Classification and Approval Guidelines
+
+This section discusses how pull requests will be classified, reviewed, and approved.
+
+## 8.1 PR Classification
+
+Pull requests shall be classified into one of the following complexity categories:
+* Model change - bug fix – change to existing logic (without major redesign) to cause it to implement the original intended behaviour and design; generally used to address an oversight in a previous contribution.
```
[the remark this answers is not in the document]
Comment (brianlynn2) [reply GOVERNANCE.md:186]: we had agreed "bug fix" at the last meeting, so let's change to that.

File: GOVERNANCE.md [hunk 7]
```diff
@@ -54,3 +54,204 @@ Inspired by [ANSI’s Essential Requirements for Due Process](https://share.ansi
...
+
+## 9.0 Release Build Approval Guidelines
+
+This section covers scheduling of minor, development, and patch releases, and approvals for all builds and releases.
+
+## 9.1 Dev Release Scheduling and Approvals
+* Dev releases may be scheduled by the maintainers  to optimize development resources, based on the queue of approved PRs
+  * There is no particular desired/expected release frequency; releases may be cut as soon as there is an approved PR, or several PRs may be consolidated into a single release at the convenience of the maintainers and dev staff
+  * *Rationale:*  Dev builds are expected to change in functionality, and getting changes out as quickly as practical is usually desirable.
+  * Each Dev Release shall require the approval of one maintainer once all the PRs are approved, and the test cass all pass successfully.
```
[the remark this answers is not in the document]
Comment (brianlynn2) [reply GOVERNANCE.md:221]: agreed

File: GOVERNANCE.md [hunk 8]
```diff
@@ -54,3 +54,204 @@ Inspired by [ANSI’s Essential Requirements for Due Process](https://share.ansi
...
+| --------------------| --------------------|  ---------------------------------------------|
+| Model - Bug fix     |1 maintainer – separate from the submitter, preferably from a separate organization | 2 maintainers; must have been reviewed by the CRWG,; if for a production version, SWG must approve; only used for recently introduced functionality  |
+| Model - Enhancement | 2 maintainers; must have been approved by a WG or the CRWG | 2 maintainers; must be on roadmap or approved by SWG; must have been approved by a WG or the CRWG; must go into a dev version; at least one maintainer must be from a separate organization   |
+| Technical - e.g. dependency update, change to mapping, reference data, documentation, changes to samples…. | At least one;  additional review up to the maintainer’s discretion – e.g. might need to consult the Technology Architecture Working Group (TAWG) | Must be approved by the TAWG; must go into a dev version |
+
+
+
+
+
+## 9.0 Release Build Approval Guidelines
```
[the remark this answers is not in the document]
Comment (brianlynn2) [reply GOVERNANCE.md:213]: agreed, thx

File: GOVERNANCE.md [hunk 9]
```diff
@@ -54,3 +54,204 @@ Inspired by [ANSI’s Essential Requirements for Due Process](https://share.ansi
...
+* Minor production releases to introduce enhancements should be combined  to minimize the number of production releases, targeting minor production releases to be issued around four weeks or so as long as there is a queue of approved PRs.  (This frequency can be increased in times of urgent need for new functionality).
+  * *Rationale:*   Minimizing the number of production releases will help with supportability, by reducing the number of releases that end users wishing to remain current need to consider, and reducing communications overhead.
+* Each Minor Production Release shall require the approval of two maintainers.
+* Minor Production releases shall be reported in brief to the CRWG and the SWG, 
+* A roadmap of anticipated minor releases shall be reported by the maintainers to the CRWG based on PRs that are in process.
+
+## 9.4  Production Patch Release Scheduling and Approvals
+
+* Patch releases to correct defects without releasing new functionality may be scheduled by the maintainers based on the presence of approves defect correction PRs, or other non-functional PRs (e.g. security remediations).
```
[the remark this answers is not in the document]
Comment (brianlynn2) [reply GOVERNANCE.md:244]: agreed

File: GOVERNANCE.md [hunk 10]
```diff
@@ -54,3 +54,204 @@ Inspired by [ANSI’s Essential Requirements for Due Process](https://share.ansi
...
+*  Patch releases require the approval of one maintainer
+*  Patch releases shall be reported to the CRWG.
+
+ ## 9.5 Summary of Release Approval Requirements
+
+ | Type of Release        | Approval Requirement |  Notes                    |
+| ----------------------- | -------------------- |  -------------------------------------------  |
+| Development Release     | 1 maintainer         | Scheduling is up to the maintainer            |
```
[the remark this answers is not in the document]
Comment (brianlynn2) [reply GOVERNANCE.md:252]: agreed

File: GOVERNANCE.md (hunk 1 above)
Comment (brianlynn2) [reply GOVERNANCE.md:248]: agreed

File: GOVERNANCE.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (chrisisla) [reply GOVERNANCE.md:145]: Okay cool, lets stick with "bug fix", thanks!

**Review by brianlynn2 — APPROVED** [review]
looks good to me

File: GOVERNANCE.md [hunk 11]
```diff
@@ -214,44 +214,42 @@ Pull requests shall be tagged in GitHub as described in discussion #2789 (https:
...
+## 9.1 Development Release Scheduling and Approvals
+* Development releases may be scheduled by the maintainers  to optimize development resources, based on the queue of approved PRs
   * There is no particular desired/expected release frequency; releases may be cut as soon as there is an approved PR, or several PRs may be consolidated into a single release at the convenience of the maintainers and dev staff
-  * *Rationale:*  Dev builds are expected to change in functionality, and getting changes out as quickly as practical is usually desirable.
-  * Each Dev Release shall require the approval of one maintainer once all the PRs are approved, and the test cass all pass successfully.
-* Dev releases shall be reported in brief to the CRWG and the SWG
+  * *Rationale:*  Development releases are expected to change in functionality, and getting changes out as quickly as practical is usually desirable.
+  * Each development release shall require the approval of one maintainer once all the PRs are approved, and the test cases all pass successfully.
+* Development release shall be reported in brief to the CRWG and the SWG
```
Comment (chrisisla) [inline GOVERNANCE.md:222]: should be "releases"

File: GOVERNANCE.md [hunk 12]
```diff
@@ -214,44 +214,42 @@ Pull requests shall be tagged in GitHub as described in discussion #2789 (https:
...
 * Minor production releases to introduce enhancements should be combined  to minimize the number of production releases, targeting minor production releases to be issued around four weeks or so as long as there is a queue of approved PRs.  (This frequency can be increased in times of urgent need for new functionality).
   * *Rationale:*   Minimizing the number of production releases will help with supportability, by reducing the number of releases that end users wishing to remain current need to consider, and reducing communications overhead.
-* Each Minor Production Release shall require the approval of two maintainers.
-* Minor Production releases shall be reported in brief to the CRWG and the SWG, 
-* A roadmap of anticipated minor releases shall be reported by the maintainers to the CRWG based on PRs that are in process.
+* Each minor production release shall require the approval of two maintainers.
+* Minor production releases shall be reported in brief to the CRWG and the SWG, 
+* A roadmap of anticipated minor production release shall be reported by the maintainers to the CRWG based on PRs that are in process.
```
Comment (chrisisla) [inline GOVERNANCE.md:240]: should be "releases"

File: GOVERNANCE.md [hunk 13]
```diff
@@ -214,44 +214,42 @@ Pull requests shall be tagged in GitHub as described in discussion #2789 (https:
...
 
 ## 9.4  Production Patch Release Scheduling and Approvals
 
-* Patch releases to correct defects without releasing new functionality may be scheduled by the maintainers based on the presence of approves defect correction PRs, or other non-functional PRs (e.g. security remediations).
-*  Patch releases require the approval of one maintainer
-*  Patch releases shall be reported to the CRWG.
+* Production patch releases to correct defects without releasing new functionality may be scheduled by the maintainers based on the presence of approved defect correction PRs, or other non-functional PRs (e.g. security remediations).
+* Production patch releases require the approval of one maintainer
+* Production patch releasess shall be reported to the CRWG.
```
Comment (chrisisla) [inline GOVERNANCE.md:246]: "releasess" -> "releases"

File: GOVERNANCE.md [hunk 14]
```diff
@@ -214,44 +214,42 @@ Pull requests shall be tagged in GitHub as described in discussion #2789 (https:
...
-| ----------------------- | -------------------- |  -------------------------------------------  |
-| Development Release     | 1 maintainer         | Scheduling is up to the maintainer            |
-| Major Release (6.0.0)   | 2 maintainers         | Scheduling via SWG; Include analysis of the changes from last major release as part of the approval  |
-| Minor Release (6.1.0)   | 2 maintainers        | Scheduling is up to the maintainers, but aim to keep to around every 4 weeks and no more than fortnightly   |
-| Patch Release (6.1.1)    | 1 maintainer         | Scheduling is up to the maintainer            |
-
-
+| Type of Release      | Approval Requirement |  Notes                                        |
+| ---------------------| -------------------- |  -------------------------------------------  |
+| Development Release  | 1 maintainer         | Scheduling is up to the maintainer            |
```
Comment (chrisisla) [inline GOVERNANCE.md:252]: thought we'd be adding an example of a Development Release too e.g. (6.0.0-dev.13)

**Review by chrisisla — CHANGES_REQUESTED** [review]

File: GOVERNANCE.md [hunk 15]
```diff
@@ -141,7 +141,7 @@ This section discusses how changes to the CDM are controlled within and between
 ## 7.1 Change Control Principles
 
 * We are trying to ensure rapid, smooth, and predictable evolution of the model by controlling when and how breaking changes are introduced
-  * We want to allow changes where needed, with a defined process make those changes, to meet evolved and improved understanding of the business and technical requirements.
+  * We want to allow changes where needed, with defined process to make those changes, to meet evolved and improved understanding of the business and technical requirements.
```
[the remark this answers is not in the document]
Comment (brianlynn2) [reply GOVERNANCE.md:144]: agreed

File: GOVERNANCE.md (hunk 11 above)
Comment (brianlynn2) [reply GOVERNANCE.md:222]: agreed

File: GOVERNANCE.md (hunk 12 above)
Comment (brianlynn2) [reply GOVERNANCE.md:240]: agreed

File: GOVERNANCE.md (hunk 13 above)
Comment (brianlynn2) [reply GOVERNANCE.md:246]: agreed

File: GOVERNANCE.md (hunk 14 above)
Comment (brianlynn2) [reply GOVERNANCE.md:252]: yes, Eteri can you please do that>

File: GOVERNANCE.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (chrisisla) [reply GOVERNANCE.md:144]: "needed, with **a** defined process" - the "a" is still missing

File: GOVERNANCE.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (chrisisla) [reply GOVERNANCE.md:145]: Where we are stating "defect approval guidelines", if we are not changing this, can we reference the section this is referring to please (as there is section entitled "defect approval guidelines"

File: GOVERNANCE.md (hunk 14 above)
Comment (chrisisla) [reply GOVERNANCE.md:252]: Can we remove the first row for "Development Release" please, as we now have it at the bottom of the table instead with the example syntax too 👍

**Review by chrisisla — CHANGES_REQUESTED** [review]

File: GOVERNANCE.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (brianlynn2) [reply GOVERNANCE.md:145]: looks like it didn't get fixed as requested, will update.

**Review by chrisisla — APPROVED** [review]

**Pull request merged**